src/core/Akka.Tests/Delivery/TestConsumer.cs

```diff
@@ -73,12 +73,18 @@ private void Active()
             _log.Info("processed [{0}] [msg: {1}] from [{2}]", job.SeqNr, job.Msg.Payload, job.ProducerId);
             job.ConfirmTo.Tell(global::Akka.Delivery.ConsumerController.Confirmed.Instance);
 
-            if (EndCondition(job))
+            if (EndCondition(job) && _messageCount > 0)
             {
                 _log.Debug("End at [{0}]", job.SeqNr);
                 EndReplyTo.Tell(new Collected(_processed.Select(c => c.Item1).ToImmutableHashSet(), _messageCount + 1));
                 Context.Stop(Self);
             }
+            else if (EndCondition(job))
+            {
```

Comment:
This is the figure - if the `EndCondition` is satisfied before we've processed any messages, we're probably being recreated via the `ProducerController<T>` + `ShardRegion`, therefore we want to ignore without sending anymore `Collected` responses.